common/omv_csi.c

```diff
@@ -459,222 +437,58 @@ int omv_csi_probe(omv_i2c_t *i2c) {
-                break;
-            #endif //(OMV_HM0360_ENABLE == 1)
-
-            #if (OMV_GC2145_ENABLE == 1)
-            case GC2145_ID:
-                if (omv_csi_set_clk_frequency(OMV_GC2145_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = gc2145_init(csi);
-                break;
-            #endif //(OMV_GC2145_ENABLE == 1)
-
-            #if (OMV_GENX320_ENABLE == 1)
-            case GENX320_ID_ES:
-            case GENX320_ID_MP:
-                if (omv_csi_set_clk_frequency(OMV_GENX320_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = genx320_init(csi);
-                break;
-            #endif // (OMV_GENX320_ENABLE == 1)
-
-            #if (OMV_PAG7920_ENABLE == 1)
-            case PAG7920_ID:
-                if (omv_csi_set_clk_frequency(OMV_PAG7920_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = pag7920_init(csi);
-                break;
-            #endif // (OMV_PAG7920_ENABLE == 1)
-
-            #if (OMV_PAG7936_ENABLE == 1)
-            case PAG7936_ID:
-                if (omv_csi_set_clk_frequency(OMV_PAG7936_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = pag7936_init(csi);
-                break;
-            #endif // (OMV_PAG7936_ENABLE == 1)
-
-            #if (OMV_PAJ6100_ENABLE == 1)
-            case PAJ6100_ID:
-                if (omv_csi_set_clk_frequency(OMV_PAJ6100_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = paj6100_init(csi);
-                break;
-            #endif // (OMV_PAJ6100_ENABLE == 1)
-
-            #if (OMV_PS5520_ENABLE == 1)
-            case PS5520_ID:
-                if (omv_csi_set_clk_frequency(OMV_PS5520_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = ps5520_init(csi);
-                break;
-            #endif // (OMV_PS5520_ENABLE == 1)
-
-            #if (OMV_FROGEYE2020_ENABLE == 1)
-            case FROGEYE2020_ID:
-                if (omv_csi_set_clk_frequency(OMV_FROGEYE2020_CLK_FREQ) != 0) {
-                    return OMV_CSI_ERROR_TIM_INIT_FAILED;
-                }
-                init_ret = frogeye2020_init(csi);
-                break;
-            #endif // (OMV_FROGEYE2020_ENABLE == 1)
+        }
 
-            #if (OMV_SOFTCSI_ENABLE == 1)
-            case SOFTCSI_ID:
-                init_ret = softcsi_init(csi);
-                break;
-            #endif // (OMV_SOFTCSI_ENABLE == 1)
+        if (init_fun ==  NULL) {
+            return OMV_CSI_ERROR_ISC_UNSUPPORTED;
+        } else if (init_fun(csi) != 0) {
+            return OMV_CSI_ERROR_ISC_INIT_FAILED;
+        }
```

Comment:
Move this after setting the correct clock frequency to match the original behavior.

Reply:
Okay, I checked everything and only lepton could have issues. Let me test that sensor with this PR. All other sensors should be okay with it.

Reply:
This commit I think solves the problem: https://github.com/openmv/openmv/pull/2737/commits/d5f1701cd025a25a6ddc472cc7a6853044f8a01a

Reply:
Can confirm the commit fixes it with the 1 second delay kept.

Reply:
Yeah, it's all good otherwise for each sensor.